README.md

```diff
@@ -1,2 +1,95 @@
+
+However, everyone is welcome to use arbitrary LLM if wanted.
+
+### Tavily API key
+
+Create a free acount on https://tavily.com and get the key there.
+
+Again, everyone is welcome to use arbitrary search engines, combine them, or even do a totally different approaches!
+
+### Private key on Gnosis Chain
+
+Use your existing or create a new wallet on Gnosis Chain. 
+
+By default the script will do only very tiny bets (0.00001 xDai per market), but of course, you can contact us on the TG group above with your public key to get some free xDai.
+
+## Task
+
+Your task is to modify `predict` function in `trader/prediction.py` by any means necessary.
+
+Goal of the `predict` function is, given an `question` about the future, answer it with either `True` (the answer is `yes`), `False` (if the answer is `no`) or `None` (if the prediction failed).
+
+All the questions are guaranteed to be about the future and to be in a binary yes/no format.
+
+You can play with the prompts, different approaches, different LLMs, search engines, or anything you can think of.
+
+The code can be messy, the only thing we ask you is for it to be reproducible on our machines, and to help with that, there is `mypy` as the only check of CI pipeline on Github.
+
+### Experimenting
+
+Run 
+
+```bash
+PYTHONPATH=. streamlit run trader/app.py
+```
+
+to start a Streamlit application where you can give your prediction method either question [from the Omen market](https://aiomen.eth.limo/), or write your own.
+
+Run 
+
+```bash
+python trader/benchmark.py --n N
+```
+
+where `N` is number of markets to do a prediction on. The benchmark script will run
+
+1. Random agent (coin flip between yes and no answers)
+2. Question-only agent (only LLM call, without any information from internet)
+3. `prediction.py/predict`-based agent
+
+on `N` open markets from https://manifold.markets. 
+
+The idea is that markets on Manifold are mostly answered by real people, so the closer your agent is to their predictions, the better. However, it isn't always the case.
+
+Bear in mind your LLM credits, Tavily credits or any other paid 3rd provider credits when running the benchmark, as it answers many markets in a single run, which can be very costly.
+
+### Submission
+
+1. Run `python trader/main.py`, it will place bets on all markets that will be used for the evaluation. You can run the script multiple times, but we will always look only at the latest bet on the market from your public key.
+2. Open a PR against this repository with your implementation and public key used for placing bets.
+3. Make sure the CI pipeline is all green.
+
+### Evaluation
+
+1. Quantitative 
+    1. We will create N markets from the address `0xa7E93F5A0e718bDDC654e525ea668c64Fd572882` by the end of the June, and they will be resolved in roughly two weeks after the creation.
+    2. We will measure the accuracy of your agent's answers (by the last bet on each market).
```

Comment:
We need to hand in prizes on July 22nd at the latest - any chance we can close the markets sooner? Otherwise we need a different evaluation metric

Reply:
Where is that stated? Can't see it in the telegram responses

Reply:
My own experience based on 50+ hackathons :laughing: 
Let me double-check this for safety.

Reply:
Based on the TG response, it seems like it's okay to close it later?